spec/system/custom_sidebar_sections_spec.rb

```diff
@@ -50,6 +50,26 @@
     expect(sidebar).to have_section_link("My preferences")
   end
 
+  it "allows the user to create custom section with /pub link" do
+    sign_in user
+    visit("/latest")
+    sidebar.click_add_section_button
+
+    expect(section_modal).to be_visible
+    expect(section_modal).to have_disabled_save
+    expect(sidebar.custom_section_modal_title).to have_content("Add custom section")
+
+    section_modal.fill_name("My section")
+
+    section_modal.fill_link("Published Page", "/pub/test")
+    expect(section_modal).to have_enabled_save
+
+    section_modal.save
+
+    expect(sidebar).to have_section("My section")
+    expect(sidebar).to have_section_link("Published Page")
```

Comment:
Up to you, but you are testing too many things here IMO. What we are testing here is that we can add a "/pub/xxx" URL, you don't need lines like:

```
expect(section_modal).to be_visible
expect(section_modal).to have_disabled_save
expect(section_modal).to have_enabled_save
```

If this doesn't work, the test will fail at the end anyways as the sidebar won't have the section. You probably have other tests checking that when nothing is filled save is disabled for example.

Reply:
Note that however, this kind of assertions can sometimes be needed in system specs, to make them more reliable, you can just use find in this case.